cerberus-web/src/main/java/com/nike/cerberus/controller/authentication/AwsIamStsAuthController.java

```diff
@@ -67,24 +68,30 @@ public AuthTokenResponse authenticate(
     String iamPrincipalArn;
     AuthTokenResponse authResponse;
 
-    try {
-      if (headerAuthorization == null || headerXAmzDate == null) {
-        throw new ApiException(DefaultApiError.MISSING_AWS_SIGNATURE_HEADERS);
-      }
-
-      AwsStsHttpHeader header =
-          new AwsStsHttpHeader(headerXAmzDate, headerXAmzSecurityToken, headerAuthorization);
-      GetCallerIdentityResponse getCallerIdentityResponse = awsStsClient.getCallerIdentity(header);
-      iamPrincipalArn = getCallerIdentityResponse.getGetCallerIdentityResult().getArn();
+    for (int count = 0; ; count++) {
```

Comment:
So this will just retry until it succeeds?

Reply:
No it will retry 5 times and then throw the exception if there is one.

Reply:
Could you make it back off exponentially for a while instead of 5 times really fast?

Reply:
I guess it depends on why it fails...